### ISSUE TITLE
Guid properties Bulk-Inserted into SQLite table are stored as lower-case strings

### ISSUE THREAD

**Issue by mprigge** [zzzprojects/EntityFramework-Extensions#355]
### Description
I am attempting to use the EFE BulkInsert function to populate a table stored in SQLite. This appears to work as expected, but I found that I was subsequently unable to retrieve the inserted records via the EF context using common linq functions like `.Single()` which depended upon the value of a Guid column. 

After a bunch of chasing dead ends, I finally determined that the current iteration of EF Core's SQLite implementation expects Guid properties to be stored as upper-case string-formatted Guid in a TEXT column (previous iterations used a byte array in a BLOB -- there's some discussion of this in [efcore/15078](https://github.com/dotnet/efcore/issues/15078)). 

The BulkInsert function is adding these rows correctly _except_ that it is storing the Guid as a lower-case string. When EF functions attempt to perform a search, they come up empty as they are expecting upper-case values.

I have attempted to work around this by specifying a value converter in `OnModelCreating` (specifying `.HasConversion()` as below), but this results in the exception quoted below. That's not my primary problem, but it is preventing me from using context value converters to work around it. If I have the converter in place, I can read the entities BulkInsert created previously (with the lower-case Guids), but BulkInsert will not function when the converter is present.

### Exception when value converter is specified

```
modelBuilder.Entity<EntityWithGuidProperty>() .Property(e => e.GuidProperty).HasConversion(
                    g => g.ToString(),
                    s => s == default ? new Guid() : new Guid(s));
```

```
System.InvalidCastException: Invalid cast from 'System.String' to 'System.Guid'.
   at System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
   at System.String.System.IConvertible.ToType(Type type, IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType)
   at Microsoft.EntityFrameworkCore.Storage.ValueConversion.ValueConverter`2.Sanitize[T](Object value)
   at Microsoft.EntityFrameworkCore.Storage.ValueConversion.ValueConverter`2.<>c__DisplayClass3_0`2.<SanitizeConverter>b__0(Object v)
   at ?.?`1.?(? ?)
   at ?.Read()
   at ?.?()
   at ?.?(? ?, DbCommand ?)
   at ?.?(? ?, DbCommand ?)
   at ?.Execute(List`1 actions)
   at ?.?(List`1 ?)
   at Z.BulkOperations.BulkOperation.Execute()
   at Z.BulkOperations.BulkOperation.BulkInsert()
   at ?.BulkInsert[T](DbContext this, IEntityType entityType, IEnumerable`1 list, Action`1 options, Boolean forceSpecificTypeMapping)
   at ?.BulkInsert[T](DbContext this, IEnumerable`1 entities, Action`1 options)
   at DbContextExtensions.?`1.?()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
```

### Further technical details
- EF version: 3.1.7
- Database Provider: SQLite

**Comment by JonathanMagnan** [comment]
Hello @mprigge ,

Thank you for reporting, we will look at it.

**Comment by JonathanMagnan** [comment]
Hello @mprigge ,

A new version is now available.

Starting for EF Core 3, the GUID should now always be in upper case as you reported.

Let us know if everything is now working.

Best Regards,

Jon

**Comment by JonathanMagnan** [comment]
Hello @mprigge ,

Since our last conversation, we haven't heard from you.

Did you get the chance to try the new version?

Don't hesitate to contact us for any questions, issues or feedback.

Best regards,

Jon

**Comment by mprigge** [comment]
Hi @JonathanMagnan! I apologize for not responding. I needed to find a solution quickly, so I ended up approaching the problem using a different extension. I am not sure when I will have an opportunity to loop back around to testing it since the changes have already shipped. However, if you're sure you've made the change you describe, I'm relatively certain that would have addressed the problem I ran into.
